lib/packaging/paths.rb

```diff
@@ -101,29 +101,85 @@ def link_name(nonfinal = false)
+    when 'deb'
+      debian_code_name = Pkg::Platforms.get_attribute(platform_tag, :codename)
+      return File.join(prefix, debian_code_name, apt_repo_name(is_nonfinal))
+    when 'dmg'
+      return File.join(prefix, 'mac', repo_name(is_nonfinal))
+    when 'msi'
+      return File.join(prefix, platform_name, repo_name(is_nonfinal))
     when 'rpm'
-      [File.join(path_prefix, yum_repo_name(nonfinal)), link_name(nonfinal).nil? ? nil : File.join(path_prefix, link_name(nonfinal))]
+      return File.join(prefix, yum_repo_name(is_nonfinal))
     when 'swix'
-      [File.join(path_prefix, platform, repo_name(nonfinal)), link_name(nonfinal).nil? ? nil : File.join(path_prefix, platform, link_name(nonfinal))]
+      return File.join(prefix, platform_name, repo_name(is_nonfinal))
+    when 'svr4', 'ips'
+      return File.join(prefix, 'solaris', repo_name(is_nonfinal))
+    else
+      raise "Error: Unknown package format '#{package_format}'"
+    end
+  end
+
+  # Construct a platform-dependent link path
+  def construct_link_path(path_data)
+    package_format = path_data[:package_format]
+    prefix = path_data[:prefix]
+    platform_name = path_data[:platform_name]
+    platform_tag = path_data[:platform_tag]
+    link = path_data[:link]
+
+    return nil if link.nil?
+
+    case package_format
+    when 'rpm'
+      return File.join(prefix, link)
+    when 'swix'
+      return File.join(prefix, platform_name, link)
     when 'deb'
-      [File.join(path_prefix, Pkg::Platforms.get_attribute(platform_tag, :codename), apt_repo_name(nonfinal)),
-       link_name(nonfinal).nil? ? nil : File.join(path_prefix, Pkg::Platforms.get_attribute(platform_tag, :codename), link_name(nonfinal))]
+      debian_code_name = Pkg::Platforms.get_attribute(platform_tag, :codename)
+      return File.join(prefix, debian_code_name, link)
     when 'svr4', 'ips'
-      [File.join(path_prefix, 'solaris', repo_name(nonfinal)), link_name(nonfinal).nil? ? nil : File.join(path_prefix, 'solaris', link_name(nonfinal))]
+      return File.join(prefix, 'solaris', link)
     when 'dmg'
-      [File.join(path_prefix, 'mac', repo_name(nonfinal)), link_name(nonfinal).nil? ? nil : File.join(path_prefix, 'mac', link_name(nonfinal))]
+      return File.join(prefix, 'mac', link)
     when 'msi'
-      [File.join(path_prefix, platform, repo_name(nonfinal)), link_name(nonfinal).nil? ? nil : File.join(path_prefix, platform, link_name(nonfinal))]
+      return File.join(prefix, platform, link)
     else
-      raise "Not sure where to find packages with a package format of '#{package_format}'"
+      raise "Error: Unknown package format '#{package_format}'"
     end
   end
 
+  # Given platform information, create symlink target (base_path) and link path in the
+  # form of a 2-element array
+  def artifacts_base_path_and_link_path(platform_tag, prefix = 'artifacts', is_nonfinal = false)
+    platform_name, version, architecture = Pkg::Platforms.parse_platform_tag(platform_tag)
```

Comment:
[nitpick] It doesn't look like we use the version or architecture from `path_data` in either of the methods, may make sense to get rid of those and change this line to
```suggestion
    platform_name, _ = Pkg::Platforms.parse_platform_tag(platform_tag)
```

Feel free to ignore :)

Reply:
Great catch. Updated.